### DISCUSSION TITLE
Feature request: Back button

### DISCUSSION THREAD

**Discussion by Nazze** [Clooos/Bubble-Card#282] [Feature requests]
Hi

I'd love to have the possible to have a back button in popups. I have a popup that has elements, that can call other popups ("nested popups", so to say). I don't want to close the popup, because I need to navigate to the previous popup again. 
Therefore, I would like to have back button (enabled via an option), that is beautifully integrated in the popup window.

Currently I have solved this with a mushroom chip card (back button variant), which works. But it looks a little bit odd and it is a little bit to small. See Screenshot..

![Bildschirmfoto 2023-09-09 um 17 00 38](https://github.com/Clooos/Bubble-Card/assets/33813707/08ed6ce8-07e5-465b-b974-cea0e6226b26)

KR
Nazze

**Comment by QuentinBtd** [comment]
To go one step further, it would even be nice to be able to choose the action of the "Close" button. 😄 

For example, in all "templates" cards of [lovelace-mushroom](https://github.com/piitaya/lovelace-mushroom), you can choose the action for "Short press", "Double press" and "Long press". I use it to open popup with the "Navigate" action.

**Comment by mleiber** [comment]
Along the same lines. The ``close`` button seems to not 'clean' the return stack. Suppose you have a subview on which you call-up a pop-up. Then close the pop-up via the ``x`` button. 

You now have to press the back-key/button of the subview twice to return to the original view. 

If instead of closing the pop-up via the ``x`` button you use the back-key then the return from the subview works as expected.

**Comment by Nazze** [comment]
Hi!
Beta 5 is just awesome @Clooos ! I like the sub buttons. The navigation URL works for me. But if you could add a “back” option addition, it would really ease my configuration work. 

Thanks for this awesome piece of software 😀

**Comment by julien94800** [comment]
Hello, did anyone found a nice way to achieve this? I'm looking for a way to have a back button just next to the close button too

**Comment by makerglenn** [comment]
...I'm just getting back to my Bubble dashboard and I have nested popups. Is there a way to return to the previous popup and not the root card? I currently coded a "back" button and navigate to the hash of the previous pop up to get back there on tap. Maybe I'm missing something or a better work around....  like Bubble card maintaining a popup stack and having a close option that backs up thru the popup stack. Would love to hear how anyone is coding this to handle this case. Thanks...

**Comment by Clooos** [comment]
Hi! It was about time! 😄

<img width="824" height="291" alt="image" src="https://github.com/user-attachments/assets/7b66615e-d175-4481-926a-73fe82223990" />
